:seedling: security: pin GitHub Actions to commit hashes

harekrishnarai opened the pull request:
#### What

This pull request pins all GitHub Actions in workflow files to specific commit hashes to improve security and ensure reproducible builds. kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What

This pull request pins all GitHub Actions in workflow files to specific commit hashes to improve security and ensure reproducible builds. is the current behavior?

#### What

This pull request pins all GitHub Actions in workflow files to specific commit hashes to improve security and ensure reproducible builds. is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.

Fixes #

or

NONE
-->

#### Special notes for your reviewer

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below.
If the PR requires additional action from users switching to the new release,
include the string "ACTION REQUIRED".

For more information on release notes see: https://git.k8s.io/release/cmd/release-notes/README.md
-->

```release-note

```

spencerschrock reviewed (dismissed):
I need to double check the source, but we intentionally pin to a tag here due to a requirement from the slsa-verifier. All of our other action usages are pinned.

spencerschrock commented:
Also can you fix DCO and ensure any follow-up commits follow the practice?
https://github.com/ossf/scorecard/runs/44897673400

harekrishnarai commented:
**Issues resolved:**

1. **DCO Fixed**: All commits now include proper DCO sign-off with 'Signed-off-by' lines
2. **Updated to v2.7.1**: Updated slsa-framework/slsa-verifier/actions/installer from v2.7.0 to v2.7.1 using commit hash ea584f4502babc6f60d9bc799dbbb13c1caa9ee6
3. **Removed date comment**: Removed the '# on 2025-06-27' comment as suggested

The PR now addresses all review feedback and is ready for re-review. Thank you for the guidance!
